### DISCUSSION TITLE
Checking the color of a CLOG object.

### DISCUSSION THREAD

**Discussion by kaeland** [rabbibotton/clog#38] [Q&A] [answered]
Hi Dr. Botton, 

I hope all is well, and thank you for building CLOG. I'm beginning the journey of learning this tool, and I really hope that it helps others build great applications! As a novice in both common lisp & CLOG, I have a question about checking the color of a clog object. For instance, after running the code below, is there a way to toggle the color of the `h1` tag? It looks like retrieving the color attribute of a clog object returns the RGB value of the object, and not its CSS namesake which may be challenging to match in a conditional like an if-statement. Do you have any advice for how to toggle the color of this object, or advice for how to approach this in general? 

```
(defpackage #:web
  (:use #:cl #:clog)
  (:export start-web-app))

(in-package :web)

(defun my-on-click (obj)
  (let ((obj-color (color (connection-data-item obj "changer"))))
    ;; Although the "obj-color" is a string, it is the
    ;; RGB value of the color property. I was expecting it to be the value  
    ;; I set it to: "purple". Is there a way to
    ;; conditionally match the color in the if-statement below?
    ;; Or, is there a CLOG color util to help find the rgb value of a
    ;; CLOG object? This would help with matching in the if-statment.
    
    (print obj-color)
    (if (string-equal obj-color "purple")
        (setf (color (connection-data-item obj "changer")) "black")
        (setf (color (connection-data-item obj "changer")) "purple"))))

(defun on-new-window (body)
  "On-new-window handler."
  (setf (title (html-document body)) "A simple App")
  (set-on-click (create-section body :h1 :content "Purple-o-nator")
                #'my-on-click)
  (setf (connection-data-item body "changer")
        (create-section body :h1 :content "Am I purple, yet?!"))
  (run body))

(defun start-web-app ()
  "Start the App!"
  (initialize #'on-new-window)
  (open-browser))
```

Thank you for such a great library, 
Kaeland

**Comment by rabbibotton** [comment, marked as answer]
You can use the function rgb:

    (if (string-equal obj-color (rgb 128 0 128))

Although works, I'd probably keep track of the state in lisp rather the query the browser.

Great start as a CLOGer :)

**Reply by rabbibotton** [reply]
https://github.com/rabbibotton/clog/blob/main/support/disc-38.lisp

**Reply by kaeland** [reply]
Got it! Thank you! :D
